### PR TITLE
sys-boot/grub: add 2.12

### PR DESCRIPTION
**Pull request by zmedico**
* Drop patch for util/grub.d/25_bli.in which was merged upstream.
* Re-add unstable keywords for supported archs.
* Create grub-core/extra_deps.lst missing from source tarball.

### CODE REVIEW

**Comment by thesamesam** [comment]
cc @xxc3nsoredxx

File: sys-boot/grub/grub-2.12.ebuild [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+		FROM_BOOTSTRAP=1 ./autogen.sh || die
+	fi
+
+	if [[ -n ${GRUB_AUTORECONF} ]]; then
+		eautoreconf
+	fi
+
+	# Avoid error due to missing Makefile rule for extra_deps.lst:
+	#	make[3]: *** No rule to make target 'grub-core/extra_deps.lst', needed by 'syminfo.lst'.  Stop.
+	touch grub-core/extra_deps.lst || die
```
Comment (floppym) [inline sys-boot/grub/grub-2.12.ebuild:189]: Please reproduce the file from the git repo. It currently contains a single line.
```
depends bli part_gpt
```

**Review by floppym — CHANGES_REQUESTED** [review]

File: sys-boot/grub/grub-2.12.ebuild (hunk 1 above)
Comment (zmedico) [reply sys-boot/grub/grub-2.12.ebuild]: Got it, thanks!

Comment (xxc3nsoredxx) [reply sys-boot/grub/grub-2.12.ebuild]: You're not far off. I sent this to GRUB yesterday when trying to build the release: https://lore.kernel.org/grub-devel/ZYPmj-z84DpRk7OQ@dj3ntoo/t/#u

It is also not the first issue which was found with the commit that introduced `extra_deps.lst`: https://lore.kernel.org/grub-devel/20231206091906.3da4f6b5.olaf@aepfle.de/t/#u But it turns out the first fix was not enough.

**Review by floppym — APPROVED** [review]

**Pull request merged**